### ISSUE TITLE
paused DRBD sync causes flood of requests to noded

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#848]
Originally reported of Google Code with ID 792.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".
gnt-cluster (ganeti v2.10.1) 2.10.1
Software version: 2.10.1
Internode protocol: 2100000
Configuration format: 2100000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.10.1
hspace (ganeti) version v2.10.1
compiled with ghc 6.12
running on linux x86_64

<b>What distribution are you using?</b>
Debian Squeeze

<b>What steps will reproduce the problem?</b>
1. Initiate a replace-disks operation.
2. Pause the DRBD sync operation with `drbdsetup {device} pause-sync`.

<b>What is the expected output? What do you see instead?</b>
To try to quickly recover from a single-point-of-failure condition, about 8 replace-disks operations were kicked off at once. This put excess load on the cluster and it was decided to pause all but one of the DRBD syncs to reduce the load. However, the noded processes then started putting a large load on the node receiving the DRBD sync streams. The node-daemon.log was showing a higher-than-usual requests per second to /blockdev_getmirrorstatus and many corresponding lvs command executions.

<b>Please provide any additional information below.</b>
Notice in the attached job info file that once the "time remaining" estimate from DRBD goes to "no time estimate", the rate at which the job is updated increases dramatically.

I suspect this function may be relevant (http://git.ganeti.org/?p=ganeti.git;a=blob;f=lib/cmdlib/instance_storage.py;hb=6e684281c5fa65261d265e89190407a7ea4f6182#l1199). The sleep on line 1265 depends on the max_time variable, which is initialized to 0 on line 1219, but only updated on line 1247 if mstat.estimated_time is not None. Perhaps when DRBD syncs are paused mstat.estimated_time is set to None and max_time remains set to 0 preventing the sleep between each loop.

```
Originally added on 2014-04-07 22:38:11 +0000 UTC.

Attachments: 
[job.675873](https://storage.googleapis.com/google-code-attachments/ganeti/issue-792/comment-0/job.675873)
[node-daemon.log.snippet](https://storage.googleapis.com/google-code-attachments/ganeti/issue-792/comment-0/node-daemon.log.snippet)

**Comment by Ganeti-Issues-Migrator** [comment]
```
Thanks for the report. I was able to reproduce this and I'm working on a fix right now.
```
Originally added on 2014-04-08 08:14:49 +0000 UTC.

Added Labels: Type-Defect Priority-Medium Component-storage
Changed State: Started
Added to Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed with

commit c1b512ca4683a917733f8ad4f9904373e47009ec
Author: Thomas Thrainer <thomasth@google.com>
Date:   Tue Apr 8 10:15:36 2014 +0200

    Fix request flooding of noded during disk sync
    
    If a user manually pauses the DRBD sync, Ganeti started to poll the node
    daemon without any waiting between request. This caused heavy load on
    the node. This patch fixes this by waiting at least 5 secondes between
    two polls if no sensible estimate for the remaining sync time could be
    made.
    
    Fixes issue 792
    
    Signed-off-by: Thomas Thrainer <thomasth@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>
```
Originally added on 2014-04-10 08:39:06 +0000 UTC.

Mentioned Issue with GoogleCode ID 792 has been migrated to GitHub with ID #848.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released in 2.10.3.
```
Originally added on 2014-04-16 10:37:20 +0000 UTC.

Changed State: Released
